pkg/operator/controllers/checker/serviceprincipalchecker_test.go

```diff
@@ -61,18 +105,45 @@ func TestServicePrincipalValid(t *testing.T) {
 		if tt.aroCluster != nil {
 			arocli = arofake.NewSimpleClientset(tt.aroCluster)
 		}
+		if tt.secret != nil {
+			kubernetescli = fake.NewSimpleClientset(tt.secret)
+		}
+
+		controller := gomock.NewController(t)
+		aad := mock_aad.NewMockAadTokenClient(controller)
+		dynamicController := gomock.NewController(t)
+		dynamic := mock_dynamic.NewMockDynamic(dynamicController)
+
+		if tt.secret != nil {
+			aadCall := aad.EXPECT().GetToken(ctx,
+				log,
+				string(tt.secret.Data["azure_client_id"]),
+				string(tt.secret.Data["azure_client_secret"]),
+				string(tt.secret.Data["azure_tenant_id"]),
+				"https://login.microsoftonline.com/",
+				"https://management.azure.com/").MaxTimes(1).Return(nil, nil)
+
+			dynamic.EXPECT().ValidateServicePrincipal(ctx,
+				string(tt.secret.Data["azure_client_id"]),
+				string(tt.secret.Data["azure_client_secret"]),
+				string(tt.secret.Data["azure_tenant_id"]),
+			).MaxTimes(1).After(aadCall).Return(nil)
+		}
 
 		sp := &ServicePrincipalChecker{
-			arocli:        arocli,
-			kubernetescli: kubernetescli,
+			log:                      log,
+			arocli:                   arocli,
+			kubernetescli:            kubernetescli,
+			getToken:                 aad.GetToken,
+			validateServicePrincipal: dynamic,
 		}
 
 		t.Run(tt.name, func(t *testing.T) {
 			err := sp.Check(ctx)
```

Comment:
Could we also add either direct or indirect coverage for `updateFailedCondition()` in the service principal checker?